source/kraken/apply_disruption.cpp

```diff
@@ -387,34 +402,45 @@ struct add_impacts_visitor : public apply_impacts_visitor {
                 LOG4CPLUS_TRACE(log, "impacted vj : " << vj_uri << " not found in data. I ignore it.");
                 continue;
             }
+            if (impacted_vj.impacted_ranks.empty()) {
+                LOG4CPLUS_TRACE(log, "impacted vj : " << vj_uri << " without impacted_ranks data. I ignore it.");
+                continue;
+            }
+            const auto begin = impacted_vj.impacted_ranks.begin();
+            const auto end = std::next(begin, impacted_vj.impacted_ranks.size() - 1);
+
             nt::VehicleJourney* vj = vj_iterator->second;
             auto& new_vp = impacted_vj.new_vp;
 
             // We keep only stop time before to touch the first impacted stop time
-            bool first_sa = false;
             for (const auto& st : vj->stop_time_list) {
                 // We need to get the associated base stop_time to compare its rank
                 const auto base_st = st.get_base_stop_time();
-                // stop is ignored if its stop_point is not in impacted_stops
-                // if we don't find an associated base we keep it
-                if (base_st && impacted_vj.impacted_ranks.count(base_st->order())) {
-                    LOG4CPLUS_TRACE(log, "Ignoring stop " << st.stop_point->uri << "on " << vj->uri);
-                    if (first_sa == true) {
-                        break;
+                if (impact->severity->effect == nt::disruption::Effect::REDUCED_SERVICE
+                    && impacted_vj.impacted_ranks.count(base_st->order())) {
```

Comment:
the previous version of the code tested if `base_st != null`. You don't need this check anymore ?